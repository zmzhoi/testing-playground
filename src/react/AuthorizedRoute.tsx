import { Route, RouteProps } from 'react-router-dom';

import useUser from './hooks/useUser';

interface Props extends RouteProps {}

function AuthorizedRoute({ path, component: Component, exact = false }: Props) {
  const { authorized } = useUser();

  if (!Component) {
    return null;
  }

  return (
    <Route
      path={path}
      exact={exact}
      render={(props) => {
        if (authorized === true) {
          return <Component {...props} />;
        }

        if (authorized === false) {
          return <h1>권한 없음</h1>;
        }

        return null;
      }}
    />
  );
}

export default AuthorizedRoute;
