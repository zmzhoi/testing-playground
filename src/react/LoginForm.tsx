import React, { useState } from 'react';

interface Props {
  onSubmit: () => void;
}

function LoginForm({ onSubmit }: Props) {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  return (
    <>
      <h2>Login</h2>
      <form onSubmit={onSubmit}>
        <label>
          이메일
          <input
            type="email"
            placeholder="이메일을 입력해 주세요."
            value={email}
            onChange={({ target: { value } }) => setEmail(value)}
          />
        </label>
        <label>
          비밀번호
          <input
            type="password"
            value={password}
            onChange={({ target: { value } }) => setPassword(value)}
          />
        </label>
        <button disabled={!email || !password}>로그인</button>
      </form>
    </>
  );
}

export default LoginForm;
