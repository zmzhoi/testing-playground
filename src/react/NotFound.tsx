interface IProps {
  path: string;
}

function NotFound({ path }: IProps) {
  return (
    <>
      <h2>404 Not Found</h2>
      <p>해당 페이지를 찾을 수 없습니다.</p>
      <img alt="404" src="https://www.example.com/not-found" />
    </>
  );
}

export default NotFound;
