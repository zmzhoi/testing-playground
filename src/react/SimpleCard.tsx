interface Props {
  title: string;
  description: string;
}

function SimpleCard({ title, description }: Props) {
  return (
    <div>
      <h1>{title}</h1>
      <p>{description}</p>
    </div>
  );
}

export default SimpleCard;
