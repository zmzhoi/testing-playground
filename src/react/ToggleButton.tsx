import useToggle from './hooks/useToggle';

interface Props {
  toggle: boolean;
}

function ToggleButton({ toggle }: Props) {
  const [on, onToggle] = useToggle(toggle);

  return <button onClick={onToggle}>{on ? 'on' : 'off'}</button>;
}

export default ToggleButton;
