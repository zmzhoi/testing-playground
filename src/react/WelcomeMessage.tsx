interface Props {
  authorized: boolean;
  username?: string;
}

function WelcomeMessage({ authorized, username }: Props) {
  const welcomeMessage = authorized
    ? `환영합니다! ${username || '이름없음'}님`
    : '로그인을 해주세요.';

  return (
    <div>
      <p>{welcomeMessage}</p>
    </div>
  );
}

export default WelcomeMessage;
