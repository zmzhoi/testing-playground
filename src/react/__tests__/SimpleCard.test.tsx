import { render } from '@testing-library/react';

import Example1 from '../SimpleCard';

it('Matchs snapshot', () => {
  const title = '이것은 제목입니다.';
  const description = '이것은 설명입니다! 가나다라마바사  아자차카 타 파  하';

  const utils = render(<Example1 title={title} description={description} />);

  expect(utils.container).toMatchSnapshot();
});

it('Props should be correct', () => {
  const title = 'Title';
  const description = 'description~';

  const utils = render(<Example1 title={title} description={description} />);

  utils.getByText(title);
  utils.getByText(description);
});
