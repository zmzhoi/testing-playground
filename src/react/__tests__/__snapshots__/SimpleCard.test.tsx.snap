// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matchs snapshot 1`] = `
<div>
  <div>
    <h1>
      이것은 제목입니다.
    </h1>
    <p>
      이것은 설명입니다! 가나다라마바사  아자차카 타 파  하
    </p>
  </div>
</div>
`;
